timeseries/src/autogluon/timeseries/models/sktime/abstract_sktime.py

```diff
@@ -141,17 +141,19 @@ def _fit(
 
             self.sktime_forecaster.fit(self._to_sktime_data_frame(train_data[[self.target]]), fh=self._fh())
 
-        self._fit_index = train_data.index.copy()
+        self._fit_hash = train_data.hash_each_item()
 
     def predict(self, data: TimeSeriesDataFrame, quantile_levels: List[float] = None, **kwargs) -> TimeSeriesDataFrame:
         self._check_predict_inputs(data, quantile_levels=quantile_levels, **kwargs)
 
         if not self.sktime_forecaster:
             raise ValueError("No sktime forecaster found. Please fit the model first.")
 
-        # TODO: reconsider when to refit the model. currently we refit whenever train and
-        #  test indices are not identical
-        if not self._fit_index.equals(data.index):
+        # sktime trains one local model for each time series (item), so we need to refit if a different set of items is
+        # given for prediction. We check that train and pred items match using hash based on `timestamp` and `target`
+        # fields (`item_id` can be different as long as `timestamp` and `target` fields match)
+        data_hash = data.hash_each_item()
+        if len(self._fit_hash) != len(data_hash) or (self._fit_hash.values != data_hash.values).any():
```

Comment:
so right now if the two data frames are not exactly the same in content we refit, this will change after per-item training logic is implemented?